woocommerce/navigation
# Add pre-commit and pre-push git hooks

Fixes #11

This PR adds pre-commit and pre-push git hooks, using `husky` and `lint-staged`, as is done in `woocommerce-admin`.

It includes a `sample-test.js` file that can be removed later on once we have real unit tests in the repo.

### Testing instructions

- Check out PR's branch
- Create a new branch off of this PR's branch to test hooks

#### Testing pre-commit hook

- Make some changes to JS files that would fail linting (such as using an undefined global in the sample-test.js file)
- Stage the changes
- Commit the changes
- Verify that the commit fails due to the linting errors
- Fix the errors
- Stage the changes
- Commit the changes
- Verify that the commit now succeeds

#### Testing pre-push hook

- Make a change to `sample-test.js` that would cause the unit tests to fail
- Stage the changes
- Commit the changes -- this should pass, as the unit tests are not run on commit
- Push the changes up to the repo 
- Verify that the push fails due to the unit test failures
- Fix the unit test
- Stage the changes
- Commit the changes
- Push the changes up to the repo
- Verify that the push now succeeds
